app/controllers/items_controller.rb

```diff
@@ -1,19 +1,39 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!
-  before_action :set_wishlist_and_item, only: %i(edit update)
+  before_action :set_wishlist, only: %i(new create edit update)
+  before_action :set_item, only: %i(edit update)
+
+  def new
+    @item = @wishlist.items.new
+    authorize! :new, @item
+  rescue CanCan::AccessDenied
+    redirect_to @wishlist, alert: t("generic_error")
```

Comment:
I've thought it through. Yeah, you're right. Both rescues ultimately lead to the same thing. But shouldn't we handle them differently? I mean, we're not logging those errors, so for us this doesn't exactly matter right now, why user gets redirected. I dunno, feels like a good precaution though, to catch it here explicitly. Alas I can't think of a solid reason for doing so. I'd leave it, and refactor the whole catching `CanCan::AccessDenied` everywhere in another issue tomorrow?